blocks/header/header.js

```diff
@@ -1,247 +1,180 @@
-    nav.removeEventListener('focusout', closeOnFocusLost);
-  }
-}
+  menuButton.setAttribute("aria-expanded", false);
 
-function getDirectTextContent(menuItem) {
-  const menuLink = menuItem.querySelector(':scope > a');
-  if (menuLink) {
-    return menuLink.textContent.trim();
-  }
-  return Array.from(menuItem.childNodes)
-    .filter((n) => n.nodeType === Node.TEXT_NODE)
-    .map((n) => n.textContent)
-    .join(' ');
-}
+  // TODO update search block
+  search.classList.add("search--no-toggle");
 
-async function buildBreadcrumbsFromNavTree(nav, currentUrl) {
-  const crumbs = [];
+  // and a menu panel
+  const menuPanel = document.createElement("div");
+  menuPanel.classList.add("nav__menu", "nav__menu--closed");
 
-  const homeUrl = document.querySelector('.nav-brand a[href]').href;
+  //style the page links
+  pageLinks.classList.add("util-detail-l");
 
-  let menuItem = Array.from(nav.querySelectorAll('a')).find((a) => a.href === currentUrl);
-  if (menuItem) {
-    do {
-      const link = menuItem.querySelector(':scope > a');
-      crumbs.unshift({ title: getDirectTextContent(menuItem), url: link ? link.href : null });
-      menuItem = menuItem.closest('ul')?.closest('li');
-    } while (menuItem);
-  } else if (currentUrl !== homeUrl) {
-    crumbs.unshift({ title: getMetadata('og:title'), url: currentUrl });
-  }
+  let isPanelOpen = false;
+  let removeNavFocusTrap = null;
 
-  const placeholders = await fetchPlaceholders();
-  const homePlaceholder = placeholders.breadcrumbsHomeLabel || 'Home';
+  /**
+   * Opens the Navigation Menu.
+   * @function
+   */
+  const openNavMenu = () => {
+    isPanelOpen = true;
 
-  crumbs.unshift({ title: homePlaceholder, url: homeUrl });
+    menuButton.setAttribute("aria-expanded", true);
+    menuButton.innerHTML = NAV_MENU_ICON_CLOSE;
 
-  // last link is current page and should not be linked
-  if (crumbs.length > 1) {
-    crumbs[crumbs.length - 1].url = null;
-  }
-  crumbs[crumbs.length - 1]['aria-current'] = 'page';
-  return crumbs;
-}
+    menuPanel.classList.add("nav__menu--open");
+    menuPanel.setAttribute("aria-hidden", false);
 
-async function buildBreadcrumbs() {
-  const breadcrumbs = document.createElement('nav');
-  breadcrumbs.className = 'breadcrumbs';
-
-  const crumbs = await buildBreadcrumbsFromNavTree(document.querySelector('.nav-sections'), document.location.href);
-
-  const ol = document.createElement('ol');
-  ol.append(...crumbs.map((item) => {
-    const li = document.createElement('li');
-    if (item['aria-current']) li.setAttribute('aria-current', item['aria-current']);
-    if (item.url) {
-      const a = document.createElement('a');
-      a.href = item.url;
-      a.textContent = item.title;
-      li.append(a);
-    } else {
-      li.textContent = item.title;
+    document.body.classList.add("js-no-scroll");
+
+    // Trap focus inside menu
+    removeNavFocusTrap = createFocusTrap(nav);
+
+    document.addEventListener("keydown", handleEscapeKey);
```

Comment:
these two functions reference each other, one is always going to throw this error